src/components/SearchBox.js

```diff
@@ -0,0 +1,48 @@
+import React, { Component } from 'react';
+import Urls from '../constants/Urls';
+
+class SearchBox extends Component {
+	constructor(props) {
+		super(props);
+
+		this.state = {
+			searchTerm: '',
+		};
+	
+		this.handleTextChange = this.handleTextChange.bind(this);
+		this.handleSearchClick = this.handleSearchClick.bind(this);
+	}
+
+	handleTextChange(changeEvent) {
+		this.setState({ searchTerm: changeEvent.target.value.trim() })
+	}
+
+	handleSearchClick() {
+		if (this.state.searchTerm.trim().length === 0) {
+			return;
+		}
+
+		if (!this.props.onResults) {
+			console.error('Results are unhandled.');
```

Comment:
might be worth displaying a message to the user here, like something went wrong, please try again later, or whatever mumbo jumbo we have for that scenario

Reply:
Will do